src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -516,13 +516,14 @@ public object[] Geometry()
         /// Sets an exsisting element's pinned status
         /// </summary>
         /// <param name="pinned">Value for pin status true/false</param>
-        public void SetPinnedStatus(bool pinned)
+        public Element SetPinnedStatus(bool pinned)
         {
-            if (this.InternalElement.Pinned == pinned) return;
+            if (this.InternalElement.Pinned == pinned) return this;
 
             TransactionManager.Instance.EnsureInTransaction(Application.Document.Current.InternalDocument);
             this.InternalElement.Pinned = pinned;
             TransactionManager.Instance.TransactionTaskDone();
+            return this;
```

Comment:
not sure if we should be returning `this` or `this.InternalElement` ? `this` technically refers to the `RevitNodes.Element` class, not a Revit element ?

Reply:
InternalElement is a `Autodesk.Revit.DB.Element`, in dynamo we want to return the   `Revit.Elements.Element` which `this` is or am i missing something?

Reply:
ah yes, you're right, forgot we have to wrap all elements 👍